tests/integration/restricted/test.sh

```diff
@@ -37,7 +38,8 @@ gcloud container clusters get-credentials ${CLUSTER_NAME} --zone ${TEST_GCP_ZONE
 # Configure if the test should clean up after itself - useful for debugging
 if [ "${NOCLEANUP:-}" != "true" ]; then
   CLUSTER_CLEANUP="gcloud container clusters delete ${CLUSTER_NAME} --zone ${TEST_GCP_ZONE} --project ${TEST_GCP_PROJECT} --quiet"
-  CLEANUP="$CLUSTER_CLEANUP; $CLEANUP"
+  DISK_CLEANUP="gcloud compute disks delete $(gcloud compute disks list --format="value(name)" --filter="name~^gke-${CLUSTER_NAME_PREFIX}.*-pvc-.* AND -users:*" --project ${TEST_GCP_PROJECT}) --zone ${TEST_GCP_ZONE} --project ${TEST_GCP_PROJECT} --quiet"
```

Comment:
The -users:* filter means "don't include anything that has a user attached" - i.e., this is an in-use disk and should not be deleted. https://groups.google.com/g/gce-discussion/c/RLrwOx8fazo/m/9ve7lIdsBQAJ

One tiny pitfall with this command is that it cannot be limited to only deleting disks associated with the current test run. Disks created through the kubernetes.io/gce-pd provider do not attach any useful labels to the disk that will let us filter.

We could try switching to a new storage provisioner but in the end I don't think it matters too much. We'll only be deleting abandoned disks, in the CI project, with the gke-ds-test-restricted prefix.

Reply:
Removed this change - it doesn't belong in a test since it isn't restricted to the current run, and the namespace deletion provides enough value on its own.